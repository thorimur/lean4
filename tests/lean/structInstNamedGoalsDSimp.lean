structure Foo (α : Type) where
  a : α

structure Bar (α : Type) extends Foo α where
  b : a = a

example : Bar Nat := by refine { a := 1, .. }
-- We're checking that the goal is not of the form `{ a := 1 }.a = { a := 1 }.a`.

instance pi.Inhabited {α : Type} {β : α → Type} [∀ x, Inhabited (β x)] : Inhabited (∀ x, β x) :=
by refine { .. }; exact fun _ => default

set_option pp.all true
#print pi.Inhabited
