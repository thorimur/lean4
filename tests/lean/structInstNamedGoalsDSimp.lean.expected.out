structInstNamedGoalsDSimp.lean:7:21-7:45: error: unsolved goals
case b
⊢ 1 = 1
def pi.Inhabited : {α : Type} →
  {β : α → Type} → [inst : (x : α) → Inhabited.{1} (β x)] → Inhabited.{1} ((x : α) → β x) :=
fun {α : Type} {β : α → Type} [inst : (x : α) → Inhabited.{1} (β x)] =>
  @Inhabited.mk.{1} ((x : α) → β x) fun (x : α) => @Inhabited.default.{1} (β x) (inst x)
